server/src/main/java/io/druid/query/lookup/LookupModule.java

```diff
@@ -185,34 +194,44 @@ public LookupResourceListenerAnnouncer(
 class LookupListeningAnnouncerConfig extends ListeningAnnouncerConfig
 {
   public static final String DEFAULT_TIER = "__default";
+  private final DataSourceNameHolder dataSourceNameHolder;
   @JsonProperty("lookupTier")
   private String lookupTier = null;
```

Comment:
I suspect it was technical challenges at the time of original implementation rather than a core need. @cheddar might be able to add more light on the subject.